src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -201,8 +219,43 @@ public void trimOldSnapshots() throws Exception {
     trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
     trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
     trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
-    trimDatabases(
-        metricsDBMap, MAX_DATABASES, PluginSettings.instance().shouldCleanupMetricsDBFiles());
+
+    boolean deleteDBFiles = PluginSettings.instance().shouldCleanupMetricsDBFiles();
+    while (metricsDBMap.size() > MAX_DATABASES) {
+      Map.Entry<Long, MetricsDB> oldestEntry = metricsDBMap.pollFirstEntry();
+      if (oldestEntry != null) {
+        Long key = oldestEntry.getKey();
+        MetricsDB value = oldestEntry.getValue();
+        value.remove();
+        if (deleteDBFiles && !batchMetricsDBSet.contains(key)) {
+          value.deleteOnDiskFile();
+        }
+      }
+    }
+    // Flush any tracking batch metrics if batch metrics is disabled. Note, in order to ensure that batch metrics
+    // consumers have had at least one cycle to use any metrics they may be holding, this flush is done before
+    // re-reading the config file to update the state of the batch metrics feature.
+    if (!batchMetricsEnabled && !batchMetricsDBSet.isEmpty()) {
+      if (deleteDBFiles) {
+        for (Long timestamp : batchMetricsDBSet) {
+          if (!metricsDBMap.containsKey(timestamp)) {
+            MetricsDB.deleteOnDiskFile(timestamp);
+          }
+        }
+      }
+      batchMetricsDBSet.clear();
+    }
+    readBatchMetricsEnabledFromConf();
```

Comment:
This sounds like this call may not belong here or the name of the enclosing method `trimOldSnapshots` might be changed to something more relevant indicating that the config is re-read here ?

Reply:
This call does belong here (see the comment above it). The method name does not need to be renamed -- the same abstraction is achieved at the end. Additionally, this is complexity that cannot be avoided. The comments are here to help explain the complexity to the reader.

Reply:
Sorry, for the confusion, by not belong here, I meant in terms of the name of the method. So please change the name to reflect that ?

Reply:
Sure, will separate out into a trimOldSnapshots method and a trimOldMetricsDBFiles method.